gitpod-io/gitpod
# Add xterm as an IDE

## Description
Adds Xterm.js as a web alternative to connecting to your workspace via SSH from your terminal. The code can be found at https://github.com/gitpod-io/xterm-web-ide and the Docker image is at [ide/xterm-web](https://console.cloud.google.com/gcr/images/gitpod-core-dev/eu/build/ide/xterm-web).

## How to test
1. Open the preview environment
2. In the preferences, select notice there is no Xterm IDE
3. In [Configcat](https://app.configcat.com/), include yourself in the enable-rule for `experimentalIdes` (non-prod)
4. Wait a couple of minutes (🫖 time)
5. Reload and see the Terminal IDE magically appear
6. Create a workspace and have a blast

## Useful commands

Re-pull the IDE: 
```sh
kubectl rollout restart deployment/ide-service
```

## Current limitations

- Does not work on mobile (tried Safari on iPadOS)

## Release Notes

```release-note
NONE
```
⬆️ we can mention it after we finish the internal dog food

## Documentation


## Todo

- [x] `gp stop` closes xterm.js editor and shows the "Workspace Stopping" screen 
- [x] `gp open` works
- [x] `gp preview` works
- [x] Ensure heartbreating works
- [x] Move Docker image from DockerHub to our GCP registry

### Nice-to-haves
- [ ] Local echo [[source](https://github.com/gitpod-io/openvscode-server/blob/gp-code/main/src/vs/workbench/contrib/terminal/browser/terminalTypeAheadAddon.ts)]

## Werft options:
